### PR TITLE
ARROW-8359: [C++/Python] Enable linux-aarch64 builds

### PR DESCRIPTION
**Pull request by xhochy**
Fixes #8128

### CODE REVIEW

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
Needs `pygit2` on `aarch64` for the package upload, requested here: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/761

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
Drone needs Credentials to be setup:

```
github3.exceptions.AuthenticationFailed: 401 Requires authentication
```

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by kszucs** [comment]
The badge should work once the PR gets merged.

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-*-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py38-aarch64

**Comment by xhochy** [comment]
Python 3.8 builds time out, somehow they are much faster on conda-forge.

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py38-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py38-aarch64

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-*

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-*-aarch64

**Comment by xhochy** [comment]
All the builds passed, this is ready for a review/merge.

File: dev/tasks/conda-recipes/drone.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    - export GIT_BRANCH="{{ arrow.branch }}"
+    - export FEEDSTOCK_NAME=arrow-cpp
+    - export FEEDSTOCK_ROOT="$DRONE_WORKSPACE/arrow/dev/tasks/conda-recipes"
+    - sed -i '$ichown -R conda:conda "$FEEDSTOCK_ROOT"' /opt/docker/bin/entrypoint
+    - yum install -y git
+    - git clone --no-checkout {{ arrow.remote }} arrow
+    - pushd arrow && git fetch -t {{ arrow.remote }} {{ arrow.branch }} && git checkout FETCH_HEAD && git submodule update --init --recursive && popd
+    - mkdir -p $(pwd)/build_artifacts && chmod a+rwx $(pwd)/build_artifacts
+    - /opt/docker/bin/entrypoint $FEEDSTOCK_ROOT/drone-steps.sh $(pwd)/build_artifacts {{ queue_remote_url }} {{ task.branch }} {{ task.tag }} {% if arrow.branch == 'master' %}1{% else %}0{% endif %}
+    # TODO: Upload to anaconda and github releases
```
Comment (kszucs) [inline dev/tasks/conda-recipes/drone.yml:44]: I guess this TODO note is outdated.

Comment (xhochy) [inline dev/tasks/conda-recipes/drone.yml:44]:
```suggestion
```

Comment (xhochy) [reply dev/tasks/conda-recipes/drone.yml]: Yes, removed.

**Review by kszucs — APPROVED** [review]
LGTM, thanks Uwe for working on this!

**Pull request closed without merging**